src/main/java/de/dennisguse/opentracks/services/TrackRecordingManager.java

```diff
@@ -138,7 +138,7 @@ Pair<Track, Pair<TrackPoint, SensorDataSet>> getDataForUI() {
         return new Pair<>(track, current);
     }
 
-    public Marker.Id insertMarker(String name, String category, String description, String photoUrl) {
+    public Marker.Id insertMarker(String name, String category, String description, String photoUrl, Track.Id trackId, TrackPoint trackPoint) {
```

Comment:
Again; rather pass the location as the reference to the trackpoint will be lost anyhow.

Reply:
And for imports: it may not even exist. 
Often markers are close to the track but not actually on it.

Reply:
not possible without further stripping down of Marker, see comment above.